### PR TITLE
Fix parsing webhook message content to request body

### LINKED ISSUE #238: [BUG] Support new lines in chime message notification
**Describe the bug**
When there are multiple lines in the chime notification, the plugin fails to send the notification

**To Reproduce**
Create a chime channel and then send a notification using that channel that has a message with multiple lines

**Expected behavior**
The notification should be sent

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by zhongnansu**
### Description
- Use XContent builder to build json object for webhook request body and covert to string. This prevents escaping `\n` and keep it in the message string that gets passed
- Format the title and message body a bit
- fix the allowed config type literals in SPI plugin setting
- Add UT

### Issues Resolved
#238 
For Chime webhook we noticed that escape sequence like new line `\n` will lead to failed notification with `SearializationException` in the response from Chime webhook server.  Notice that Slack webhook feature doesn't have such issue before or after this change.
```
[2021-07-20T13:27:35,679][ERROR][o.o.n.s.f.WebhookDestinationFactory] [3c22fbe8fc7e.ant.amazon.com] Exception sending message: org.opensearch.notifications.spi.model.MessageContent@13f3b910
java.io.IOException: Failed: HttpResponseProxy{HTTP/1.1 400 Bad Request [Content-Type: application/json, Content-Length: 16, Connection: keep-alive, Date: Tue, 20 Jul 2021 20:27:35 GMT, x-amzn-ErrorType: SerializationException, x-amzn-RequestId: 2e761ad3-86a4-4efb-81f7-815baebba67c, X-Cache: Error from cloudfront, Via: 1.1 3f3347264bcaae7af741e2a2f692c6a1.cloudfront.net (CloudFront), X-Amz-Cf-Pop: SEA19-C3, X-Amz-Cf-Id: gYKFJ4ZLhRSmnJxxOg_ezECHcs_ZGKXG8su3-0gyejWEumMLCJx_tg==] ResponseEntityProxy{[Content-Type: application/json,Content-Length: 16,Chunked: false]}}
```

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dai-chen** [comment]
If not, could you convert the issue section to an issue for tracking?

File: notifications/spi/src/test/kotlin/org/opensearch/notifications/spi/ChimeDestinationTests.kt [hunk 1]
```diff
@@ -182,4 +182,17 @@ internal class ChimeDestinationTests {
             throw ex
         }
     }
+
+    @Test
+    fun `test build webhook request body for chime should have title included and prevent escape`() {
+        val httpClient = DestinationHttpClient()
+        val title = "test Chime"
+        val messageText = "line1\nline2"
```
Comment (dai-chen) [inline notifications/spi/src/test/kotlin/org/opensearch/notifications/spi/ChimeDestinationTests.kt:190]: Wondering if any other escaping character needs to test? ex. `\r`, `\t`

Comment (zhongnansu) [reply notifications/spi/src/test/kotlin/org/opensearch/notifications/spi/ChimeDestinationTests.kt]: good point, added some more common escape characters and setup paramized testing

**Comment by zhongnansu** [comment]
> If not, could you convert the issue section to an issue for tracking?

opened an issue for this

**Review by dai-chen — APPROVED** [review]
Thanks for the fix!

**Pull request merged**